funds-manager/funds-manager-server/src/custody_client/gas_wallets.rs

```diff
@@ -10,18 +11,52 @@ use rand::thread_rng;
 use tracing::info;
 
 use crate::{
+    custody_client::DepositWithdrawSource,
     db::models::GasWalletStatus,
     error::FundsManagerError,
     helpers::{create_secrets_manager_entry_with_description, get_secret},
 };
 
 use super::CustodyClient;
 
+/// The threshold beneath which we skip refilling gas for a wallet
+///
+/// I.e. if the wallet's balance is within this amount of the desired fill, we
+/// skip refilling
+pub const GAS_REFILL_TOLERANCE: f64 = 0.001; // ETH
+/// The amount to top up a newly registered gas wallet
+pub const DEFAULT_TOP_UP_AMOUNT: f64 = 0.01; // ETH
+
 impl CustodyClient {
     // ------------
     // | Handlers |
     // ------------
 
+    /// Refill gas for all active wallets
+    pub(crate) async fn refill_gas_for_active_wallets(
+        &self,
+        fill_to: f64,
+    ) -> Result<(), FundsManagerError> {
+        // Fetch all active gas wallets
+        let active_wallets = self.get_active_gas_wallets().await?;
+
+        // Filter out those that don't need refilling
+        let mut wallets_to_fill: Vec<(String, f64)> = Vec::new(); // (address, fill amount)
+        for wallet in active_wallets {
+            let bal = self.get_ether_balance(&wallet.address).await?;
+            if bal + GAS_REFILL_TOLERANCE < fill_to || bal.is_zero() {
```

Comment:
we only need the `bal.is_zero()` check in the case that `GAS_REFILL_TOLERANCE >= fill_to`, which isn't something we'd ever expect, so perhaps we can remove the `bal.is_zero()` check?